lib/rules/prefer-function-component.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @fileoverview Enforce function components over class components
+ * @author Tate Thurston
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+const Components = require('../util/Components');
+const ast = require('../util/ast');
+
+const COMPONENT_SHOULD_BE_FUNCTION = 'componentShouldBeFunction';
+const ALLOW_COMPONENT_DID_CATCH = 'allowComponentDidCatch';
+const COMPONENT_DID_CATCH = 'componentDidCatch';
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforce components are written as function components',
+      category: 'Stylistic Issues',
+      recommended: false,
+      suggestion: false,
+      url: docsUrl('prefer-function-component')
+    },
+    fixable: false,
```

Comment:
is there a reason this rule isn't fixable?

Any component with a constructor body or class methods, of course, couldn't be fixable, but the example in the readme could.

Reply:
Lots of rules are partially fixable; obv it's ideal when things are fully fixable tho.

Actually wait, I'd forgotten about prefer-stateless-function. How is this rule different from that one? https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md

Reply:
I think https://github.com/yannickcr/eslint-plugin-react/issues/2860 explains the difference well:

> The prefer-stateless-function rule is not enough for this case, since it allows class components provided they have a state or additional methods (like handlers defined as a class methods). It is not possible to forbid the use of state altogether (you can disable setState, but defining state is still allowed, silencing prefer-stateless-function), and filtering class methods using eslint rules without interfering with other code is complicated.

Reply:
i see, so this is a stronger thing than that, where prefer-stateless-function pushes SFCs when possible, this rule would ban class components entirely in favor of hook-using SFCs. Gotcha.